packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Alert from '../Alert/Alert';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  stacked,
+  striped,
+  scrollTable,
+  scrollableCaption,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [tabIndex, setTabIndex] = useState(null);
+
+  useEffect(() => {
```

Comment:
We should definitely leave comments explaining this logic and the reasoning for it (i.e. to show a message when scrollbar appears).

Reply:
Small nitpick, but you don't need to explain what a hook is or how it works, just the part about how this hook listens to the window resize event to dynamically handle horizontal scrollable tables.